### ISSUE TITLE
Cannot format large MDT with ldiskfs 

### ISSUE THREAD

**Issue by Alucardfh** [intel-hpdd/intel-manager-for-lustre#450]
### Description 

When trying to create a MDT on a 29TB volume, the mke2fs command receive ```^extents``` instead of ```extents``` which cause the command to fails.

### Repro

Create an MDT on a large LUN.

```modprobe osd_ldiskfs: 0


mkfs.lustre --mdt --mgsnode=172.21.61.200@tcp0 --mgsnode=172.21.61.206@tcp0 --failnode=172.21.61.200@tcp0 --reformat --index=0 --mkfsoptions=-I 512 -i 2048 -J size=2048 --backfstype=ldiskfs --fsname=BIGSI01 /dev/mapper/mpathb: 1

   Permanent disk data:
Target:     BIGSI01:MDT0000
Index:      0
Lustre FS:  BIGSI01
Mount type: ldiskfs
Flags:      0x61
              (MDT first_time update )
Persistent mount opts: user_xattr,errors=remount-ro
Parameters:  mgsnode=172.21.61.200@tcp:172.21.61.206@tcp failover.node=172.21.61.200@tcp

device size = 30501008MB
formatting backing filesystem ldiskfs on /dev/mapper/mpathb
	target name   BIGSI01:MDT0000
	4k blocks     7808258048
	options       -I 512 -i 2048 -J size=2048 -q -O dirdata,uninit_bg,^extents,mmp,dir_nlink,quota,huge_file,64bit,flex_bg -E lazy_journal_init -F
mkfs_cmd = mke2fs -j -b 4096 -L BIGSI01:MDT0000 -I 512 -i 2048 -J size=2048 -q -O dirdata,uninit_bg,^extents,mmp,dir_nlink,quota,huge_file,64bit,flex_bg -E lazy_journal_init -F /dev/mapper/mpathb 7808258048
   Found a gpt partition table in /dev/mapper/mpathb
   Extents MUST be enabled for a 64-bit filesystem.  Pass -O extents to rectify.


mkfs.lustre FATAL: Unable to build fs /dev/mapper/mpathb (256)

mkfs.lustre FATAL: mkfs failed 256
```

If I manually try the command and remove the "^" character, the command succeeds.

If had a look into the repo to see if I can find where this issue is coming from but I only found this issue : https://github.com/intel-hpdd/intel-manager-for-lustre/issues/439 which seems to have the same problem.

### Version

What version of IML : 4.0.2
What Operating System : Centos
What Operating System version:  7.4

**Comment by Alucardfh** [comment]
Ok found where the "^extends" come from : 

It's in build in the mkfs.lustre binary from lustre-2.10.2-1.src.rpm

Exactly it seems to come from here:
`lustre/utils/mount_utils_ldiskfs.c:601:         append_unique(anchor, ",", "^extents", NULL, maxbuflen);`

The line is the same for all the 2.10 releases, so this might not be entirely related to the issue (or else this probably would have shown in more places).

Although, I am not sure that line 601 is supposed to look like that, as line 600 which add "uninit_bg" to the options list is :

`lustre/utils/mount_utils_ldiskfs.c:600:         append_unique(anchor, ",", "uninit_bg", NULL, maxbuflen);`

**Comment by jgrund** [comment]
Hi @Alucardfh,

Given this looks like a Lustre issue, I have moved it to that project. You can follow up here:

https://jira.hpdd.intel.com/browse/LU-10520

Thanks,

Joe

**Comment by adilger** [comment]
There is no point in formatting an MDT filesystem larger than about 8-16TB, unless the new Data-on-MDT (DoM) feature is used, but that feature will not be available until the Lustre 2.11 release.

Since the MDT (without the DoM feature) only holds inodes (1KB in size for 2.10 and later) plus directories, xattrs, and some Lustre log files (average 2KB per inode), and there is an upper limit of 4B inodes, 4B * 2KB = 8TB. Having a larger MDT is largely a waste of space, since the extra space above 8TB cannot be used until the DoM feature is available.

If you are formatting this very large MDT in anticipation of the DoM feature, and are aware of this limitation that is OK. We need to make a patch to libmount_utils_ldiskfs.c to enable the extents feature only for MDT filesystems over 16TB in size.
